### ISSUE TITLE
Make 'seconds' field optional

### ISSUE THREAD

**Issue by zslayton** [zslayton/cron#13]
The `year` field is currently optional, with expressions being comprised of six or seven fields accordingly. I would like to also accept expressions which have neither `seconds` nor `years`, for a total of five fields. Importantly, expressions that **do not** include `seconds` but which **do** include `years` would be illegal.

**Comment by anton-dutov** [comment]
@zslayton 

May be use proxy object 'ScheduleBuilder'

    let s = Schedule::builder() // ScheduleBuilder::default()
        .with_seconds(true)
        .with_years(true)
        .from_str(...)

**Comment by zslayton** [comment]
> Importantly, expressions that do not include seconds but which do include years would be illegal.

To clarify this, the parser would treat any expression with 6 fields as having `years`. It would be impossible to detect the user's intent in some cases (e.g. `* * * * * *`), so this would simply be noted in the documentation.

> May be use proxy object 'ScheduleBuilder'

I'm open to this approach. There are other settings that it would be nice to support -- e.g. whether `day_of_week` should be 0-indexed or 1-indexed, which varies between implementations. (This is relevant to #37.)

**Comment by Arshia001** [comment]
@zslayton this issue is completely breaking my use-case. I can implement a fix, which shouldn't be too hard, but first, I need to ask:

> Importantly, expressions that do not include seconds but which do include years would be illegal.

I take this to mean that *any* expression with 6 fields will parse as having seconds and not years, even if it uses named values that would parse successfully if taken to be an expression with a year, such as `* */5 * * Mon *`. Is this the intent?

I don't suppose it would be too difficult to support those as well, but it may end up confusing users if we do it that way...

**Comment by zslayton** [comment]
> I take this to mean that any expression with 6 fields will parse as having seconds and not years, even if it uses named values that would parse successfully if taken to be an expression with a year, such as * */5 * * Mon *. Is this the intent?

Yes, that's what I intended. I agree there are many expressions with 6 fields that could unambiguously be parsed correctly, but I'm concerned about the ones like `* * * * * *` that cannot. 

Having a `ScheduleBuilder` type to specify configuration seems like the best way forward with this, but I'm afraid I don't have time to design/implement it. If anyone wants to take a crack at it, I suggest looking at [the `bon` crate](https://github.com/elastio/bon?tab=readme-ov-file#struct-builder) to create a builder that leverages the typestate pattern and offers sane defaults.

Off the top of my head, we would want to be able to specify whether a 6-field expression should be interpreted as having a `year` or `second` field, whether `day_of_week` should be `0`- or `1`-indexed, and what names to accept for days of the week (to allow for non-`en_US` locales).

**Comment by Arshia001** [comment]
This is all quite interesting! I can certainly take a shot later. However, the specific fix I had in mind was, in the parser:

https://github.com/zslayton/cron/blob/956beaf3cfe32091dc7a0b371340b59ae5e1a860/src/parsing.rs#L281

We can parse 5 mandatory and up to two optional fields, then decide based on the count whether to parse as without seconds (5), with seconds (6) or with years (7). This is not a complete solution, but keeps the existing behavior while enabling parsing of 5-field
